__tests__/index.js

```diff
@@ -5,19 +5,59 @@ import nock from 'nock'
 import {autocomplete, bulk, reverse, search} from '../index'
 
 const mockBulkResult = require('./mock-bulk-result.json')
+const mockBulkResultWithBadAddress = require('./mock-bulk-result-with-bad-address.json')
 const mockReverseResult = require('./mock-reverse-result.json')
 const mockSearchResult = require('./mock-search-result.json')
 const mockSuggestResult = require('./mock-suggest-result.json')
 
+function prepareNock (
+  type: 'autocomplete' | 'bulk' | 'search' | 'reverse',
+  snapshotUri = true,
+  response?: Object
+) {
+  if (type === 'bulk') {
+    // nock for auth token
+    nock('https://www.arcgis.com/')
+      .post(/sharing\/oauth2\/token/)
+      .reply(200, { access_token: 'test', expires_in: 86400 })
+  }
+
+  let n = nock('https://geocode.arcgis.com/')
+
+  switch (type) {
+    case 'autocomplete':
```

Comment:
Point here directly related to the above: is this really worth setting up instead of doing:

```js
nock(ARCGIS_URL)
  .get(SUGGEST_URL)
  .reply(200, (r) => mockSuggestResult)
```

Which leaves you with direct control over the nock object where it's being used. If you really want to keep it DRY then create smaller, more composable functions:

```js
const arcGet = path => nock(ARCGIS_URL).get(path)
...
arcGet(SUGGEST_URL).reply(200, r => mockSuggestResult)
```

Keep away from "options" objects when the logic can be decided outside of calling that function.

Reply:
Related: I misread the function signature and thought `snapshotUri` defaulted to false. If you do want to abstract out that functionality then the following would be appropriate:

```js
const arcGet = path => nock(ARCGIS_URL).get(path)
const snapshotUri = (type, response) => (uri, rb) => {
  expect(uri).toMatchSnapshot(`basic ${type} request uri`)

  return response
}
...
arcGet(SUGGEST_URL).reply(200, getUriSnapshot('autocomplete', mockSuggestResult))
```

Reply:
I like this idea of composable functions.  I'm going to try to train my brain to think of using these before or instead of the larger functions with lots of options.

Reply:
Wow, refactoring to using composable functions reduced the lines of code by 14%!  I need to get better at this.